src/main/java/org/skife/jdbi/v2/unstable/BindIn.java

```diff
@@ -90,21 +100,98 @@ public void apply(SQLStatement q) throws SQLException
     class BindingFactory implements BinderFactory<BindIn>
     {
         @Override
-        public Binder build(BindIn in)
+        public Binder build(final BindIn in)
         {
             final String key = in.value();
 
-            return new Binder<Annotation, Iterable>()
+            return new Binder<Annotation, Object>()
             {
                 @Override
-                public void bind(SQLStatement q, Annotation bind, Iterable coll)
+                public void bind(final SQLStatement q, final Annotation bind, final Object arg)
                 {
+                    final Object[] arr = Util.toArray(arg);
+
                     int idx = 0;
-                    for (Object s : coll) {
+                    for (final Object s : arr)
+                    {
                         q.bind("__" + key + "_" + idx++, s);
                     }
                 }
             };
         }
     }
+
+    final class Util
+    {
+        private Util()
+        {
+        }
+
+        static Object[] toArray(final Object obj)
```

Comment:
Consider binding a `List<MyBean>` -- right now, it'll assemble an intermediate `MyBean[]` which is only then used to iterate over to bind.  You could bind directly from the `Iterator` and remove the temporary copy.  It's a relatively small point, so it's an optional fixup if you like :)